Dockerfile

```diff
@@ -1,37 +1,33 @@
 FROM ubuntu:16.04
 
-RUN apt-get update && apt-get install -y \
-    pkg-config \
-    rabbitmq-server \
-    python-pip \
-    curl \
-    jq \
-    google-perftools \
-    capnproto \
-    git \
-    libsnappy-dev \
-    libgoogle-perftools-dev \
-    libsodium* \
-    libzmq3-dev \
-    libssl-dev \
- && rm -rf /var/lib/apt/lists \
- && rm -rf ~/.cache/pip
+COPY solc /usr/bin/
+COPY libgmssl.so.1.0.0 /usr/local/lib/
 
+RUN apt-get update \
+    && apt-get install -y \
+        pkg-config \
+        rabbitmq-server \
+        python-pip \
+        curl \
+        jq \
+        google-perftools \
+        capnproto \
+        git \
+        libsnappy-dev \
+        libgoogle-perftools-dev \
+        libsodium* \
+        libzmq3-dev \
+        libssl-dev \
+    && chmod +x /usr/bin/solc \
+    && ln -srf /usr/local/lib/libgmssl.so.1.0.0 /usr/local/lib/libgmssl.so \
+    && ldconfig \
+    && pip install -U pip ethereum==2.2.0 pysodium toml \
```

Comment:
To my knowledge, do these commands in one statement is for reducing image size.

By the way, why caching matters in this situation?

Reply:
Doing one job in one statement **with cleaning up** also can reducing image size.
The key point is not do all commands in one statement.
The key point is **doing clean up in the same statement** after the command which messes it up.

Caching is good for debug and test the docker build scripts.
If `docker build` failed at middle of statements, it will be restarted at the last successful layer, not the begin of the script.

In the near future, there will be more than one docker image for cita (such as cita-build, cita-run and so on).
If they have same base image layers, it also can reducing the disk usage and minimizing network traffic.

And, imo, splitting different jobs in different statements, the script will be more flexible.